### PR TITLE
FEATURE: Create logs for Group Moderator changes

### PR DESCRIPTION
**Pull request by jbrw**

### CODE REVIEW

File: spec/services/staff_action_logger_spec.rb [hunk 1]
```diff
@@ -546,4 +546,63 @@
...
+      expect { logger.log_topic_closed(topic, closed: false) }.to change { UserHistory.count }.by(1)
+      user_history = UserHistory.last
+      expect(user_history.action).to eq(UserHistory.actions[:topic_opened])
+    end
+  end
+
+  describe 'log_topic_archived' do
+    fab!(:topic) { Fabricate(:topic) }
+
+    it "raises an error when argument is missing" do
```
Comment (tgxworld) [inline spec/services/staff_action_logger_spec.rb:571]:
```suggestion
    it "raises an error when argument is invalid" do
```

File: spec/services/staff_action_logger_spec.rb [hunk 2]
```diff
@@ -546,4 +546,63 @@
...
+      expect { logger.log_topic_archived(topic, archived: false) }.to change { UserHistory.count }.by(1)
+      user_history = UserHistory.last
+      expect(user_history.action).to eq(UserHistory.actions[:topic_unarchived])
+    end
+  end
+
+  describe 'log_post_staff_note' do
+    fab!(:post) { Fabricate(:post) }
+
+    it "raises an error when argument is missing" do
```
Comment (tgxworld) [inline spec/services/staff_action_logger_spec.rb:589]:
```suggestion
    it "raises an error when argument is invalid" do
```

File: spec/services/staff_action_logger_spec.rb [hunk 3]
```diff
@@ -546,4 +546,63 @@
     end
 
   end
+
+  describe 'log_topic_closed' do
+    fab!(:topic) { Fabricate(:topic) }
+
+    it "raises an error when argument is missing" do
```
Comment (tgxworld) [inline spec/services/staff_action_logger_spec.rb:553]:
```suggestion
    it "raises an error when argument is invalid" do
```

File: app/models/topic.rb [hunk 4]
```diff
@@ -621,6 +621,12 @@ def update_status(status, enabled, user, opts = {})
     TopicStatusUpdater.new(self, user).update!(status, enabled, opts)
     DiscourseEvent.trigger(:topic_status_updated, self, status, enabled)
 
+    if status == 'closed'
+      StaffActionLogger.new(user).log_topic_closed(self, closed: enabled)
+    elsif status == 'archived'
+      StaffActionLogger.new(user).log_topic_archived(self, archived: enabled)
```
Comment (tgxworld) [inline app/models/topic.rb:627]: I think we should update the tests for Topic#update_status to ensure that the staff action is logged.

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -485,6 +487,9 @@ def notice
       post.delete_post_notices
     end
 
+    details = { new_raw_value: params[:notice], old_value: previous_notice }
+    StaffActionLogger.new(current_user).log_post_staff_note(post, details)
```
Comment (tgxworld) [inline app/controllers/posts_controller.rb:491]: I think we should update the request specs for this controller action to ensure that the staff action is logged too.

File: spec/services/staff_action_logger_spec.rb [hunk 6]
```diff
@@ -546,4 +546,63 @@
...
+      expect { logger.log_topic_closed(nil) }.to raise_error(Discourse::InvalidParameters)
+    end
+
+    it "creates a new UserHistory record" do
+      expect { logger.log_topic_closed(topic, closed: true) }.to change { UserHistory.count }.by(1)
+      user_history = UserHistory.last
+      expect(user_history.action).to eq(UserHistory.actions[:topic_closed])
```
Comment (tgxworld) [inline spec/services/staff_action_logger_spec.rb:560]:
```suggestion
      expect { logger.log_topic_closed(topic, closed: true) }.to change { UserHistory.where(action: UserHistory.actions[:topic_closed].count) }.by(1)
```

File: app/models/topic.rb (hunk 4 above)
Comment (jbrw) [reply app/models/topic.rb:627]: Done

File: app/controllers/posts_controller.rb (hunk 5 above)
Comment (jbrw) [reply app/controllers/posts_controller.rb:491]: Done

File: spec/services/staff_action_logger_spec.rb (hunk 6 above)
Comment (jbrw) [reply spec/services/staff_action_logger_spec.rb]: Moved the `.count` outside of the parens, but yes: done

**Pull request merged**